server/src/main/java/io/druid/server/coordinator/rules/LoadRule.java

```diff
@@ -346,13 +348,17 @@ private boolean loadingInProgress(final DruidCluster druidCluster)
   private static int dropForTier(
       final int numToDrop,
       final NavigableSet<ServerHolder> holdersInTier,
-      final DataSegment segment
+      final DataSegment segment,
+      final BalancerStrategy balancerStrategy
   )
   {
     int numDropped = 0;
 
-    // Use the reverse order to get the holders with least available size first.
-    final Iterator<ServerHolder> iterator = holdersInTier.descendingIterator();
+    final NavigableSet<ServerHolder> isServingSubset =
+        holdersInTier.stream().filter(s -> s.isServingSegment(segment)).collect(Collectors.toCollection(TreeSet::new));
```

Comment:
There's still a (now pointless) check for `isServingSegment` below. It should be removed lest it confuse people.

Reply:
I left the check in case in the time between the initial filtering and the loop something changed and the server was no longer serving the segment, but I can remove the extra check if you think it's safe.

Reply:
If there is a race between `pickServersToDrop` and this line, then there's still a race even with the check. I think it's ok to leave the check in, but maybe log a warning if it actually gets tripped. (We don't expect it to unless there is some race)

Reply:
:+1: added `log.warn`